include/silo/query_engine/bad_request.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <stdexcept>
+#include <string>
+
+#define CHECK_SILO_QUERY(condition, message) \
+   if (!(condition)) {                       \
+      throw silo::BadRequest(message);       \
+   }
```

Comment:
Actually I like the change. We just need to be conscious about the decision that this leaks the concept of having a HTTP interface into the query engine.

Initially we thought that SILO might also have other interfaces (e.g. CLI), so we kept HTTP specifics out of the query engine/database code.

I'm not saying that we shouldn't do that. AFAIK there are not plans to add additional interfaces anytime soon. I just wanted to make this consequence explicit.